src/awesome_map/BoundaryInterceptor.js

```diff
@@ -151,6 +166,39 @@ define(function(require) {
         this._easing = options.easing || EasingFunctions.easeOutQuart;
 
         /**
+         * Defines the minimum number of pixels beyond the boundary which a scroll event
+         * must reach for boundary events to be fired.
+         * @type {number}
+         * @private
+         */
+        this._boundarySensitivity = options.boundarySensitivity || DEFAULT_BOUNDARY_SENSITIVITY;
+
+        /**
+         * Observable for subscribing to scroll events beyond AwesomeMap boundaries.
+         *
+         * @method AwesomeMap#onScrollPastTopBoundary
+         * @param {Function} callback
+         *        Invoked with (sender, {
+         *            boundary: {@link BoundaryTypes}
+         *        })
+         */
+        this.onScrollPastBoundary = Observable.newObservable();
+
+        /**
+         * A list of boundaries that have been in view since the last completed user event 
+         * @type {Array.<BoundaryType>}
+         * @private
+         */
+        this._visibleBoundaries = [];
+        // Initialize with all of the boundaries, and any that are not in view will be removed
+        // when an event is processed.
+        for (var boundary in BoundaryTypes) {
+            if (BoundaryTypes.hasOwnProperty(boundary)) {
```

Comment:
Eh, technically yes it isnt essential, but i like things to be explicit as possible.  I think it's probably best to introduce object properties in the constructor even if they aren't used right away (or even possibly at all) rather than just attaching them randomly elsewhere in the code which might be harder to track down.  It also provides a good place to look for documentation.